### PR TITLE
fix(transport): add ShouldUseMTLSEndpoint(...)

### PR DESCRIPTION
**Pull request by shinfan**
Create ShouldUseMTLSEndpoint(...) which returns if the MTLS endpoint should be used. This method is needed for Terraform to onboard mTLS and we do not want to Terraform to write its own logic to detect which endpoint to use.

### CODE REVIEW

File: transport/cert/dca.go [hunk 1]
```diff
@@ -43,7 +42,7 @@ const (
 // GetClientCertificateSourceAndEndpoint is a convenience function that invokes
 // getClientCertificateSource and getEndpoint sequentially and returns the client
 // cert source and endpoint as a tuple.
-func GetClientCertificateSourceAndEndpoint(settings *internal.DialSettings) (cert.Source, string, error) {
+func GetClientCertificateSourceAndEndpoint(settings *internal.DialSettings) (Source, string, error) {
```
Comment (codyoss) [inline transport/cert/dca.go:45]: I don't think this package can be made public as is. The input to this function is a struct that can't be made public. I think for now a little bit of copying is the right answer. This change would add quite a bit of public surface.

**Review by codyoss — CHANGES_REQUESTED** [review]

**Comment by shinfan** [comment]
Thanks for review. PTAL. @codyoss 

Not related to this PR but I am a little concerned about the package name as I feel we should have the top-level package called "mtls" instead of "cert" to be more future proof, since we will add some more mtls related features later. Is this something too late to change?

**Comment by codyoss** [comment]
What is the use case where these methods need to be exposed? We already allow for returning a `http.Client` that uses the certSource to setup the TLSConfig. I guess I am lacking the proper context here, is there a design doc that can be shared with me? Are other languages exposing similar features?

**Comment by codyoss** [comment]
Closing this PR for now as we decided to take a different direction.

**Pull request closed without merging**